File: Sources/IO/Handlers/HandleBackspace.swift
func handleBackspace(_ sequence: String, _ prompt: inout PromptState, _ session: borrowing SessionState) -> InputAction {
    if !prompt.content.isEmpty {
        prompt.content.removeLast()
        print("\u{1B}[1D\u{1B}[K", terminator: "")
    }

 	return InputAction.continueReading
}

File: src/io/handlers/handle_backspace.rs
use crate::models::prompt_state::PromptState;
use crate::core::session_state::SessionState;
use crate::io::handler_registry::InputAction;

pub fn handle_backspace(_sequence: &str, prompt: &mut PromptState, _session: &SessionState) -> InputAction {
    if !prompt.content.is_empty() {
        prompt.content.pop();
        print!("\x1B[1D\x1B[K");
    }
    InputAction::ContinueReading
}
